command/update/command.go

```diff
@@ -117,112 +113,48 @@ func (c *Command) Execute(cmd *cobra.Command, args []string) {
 func (c *Command) execute() error {
 	var err error
 
+	var k8sClient kubernetes.Interface
+	{
+		k8sConfig := k8s.DefaultConfig()
+
+		k8sConfig.Address = f.Kubernetes.Address
+		k8sConfig.Logger = microloggertest.New()
+		k8sConfig.InCluster = f.Kubernetes.InCluster
+		k8sConfig.TLS.CAFile = f.Kubernetes.TLS.CaFile
+		k8sConfig.TLS.CrtFile = f.Kubernetes.TLS.CrtFile
+		k8sConfig.TLS.KeyFile = f.Kubernetes.TLS.KeyFile
+
+		k8sClient, err = k8s.NewClient(k8sConfig)
+		if err != nil {
+			return microerror.MaskAny(err)
+		}
+	}
+
 	// At first we have to sort out which provider to use. This is based on the
 	// flags given to the updater.
 	var newProvider provider.Provider
 	{
-		k := f.Provider.Kind
-		switch k {
-		case bridge.Kind:
-			if len(f.Updater.Pod.Names) != 1 {
-				return microerror.MaskAnyf(invalidConfigError, "bridge provider expects 1 pod name")
-			}
-			podName := f.Updater.Pod.Names[0]
+		bridgeConfig := bridge.DefaultConfig()
 
-			bridgeConfig := bridge.DefaultConfig()
-			bridgeConfig.BridgeName = f.Provider.Bridge.Name
-			bridgeConfig.Logger = c.logger
-			bridgeConfig.PodName = podName
-			newProvider, err = bridge.New(bridgeConfig)
-			if err != nil {
-				return microerror.MaskAny(err)
-			}
-		case env.Kind:
-			envConfig := env.DefaultConfig()
-			envConfig.Logger = c.logger
-			envConfig.PodNames = f.Updater.Pod.Names
-			envConfig.Prefix = f.Provider.Env.Prefix
-			newProvider, err = env.New(envConfig)
-			if err != nil {
-				return microerror.MaskAny(err)
-			}
-		case etcd.Kind:
-			var storageService microstorage.Service
-			{
-				storageConfig := microstorage.DefaultConfig()
-				storageConfig.EtcdAddress = f.Provider.Etcd.Address
-				storageConfig.EtcdPrefix = f.Provider.Etcd.Prefix
-				storageConfig.Kind = f.Provider.Etcd.Kind
-				storageService, err = microstorage.New(storageConfig)
-				if err != nil {
-					return microerror.MaskAny(err)
-				}
-			}
+		bridgeConfig.Logger = c.logger
 
-			etcdConfig := etcd.DefaultConfig()
-			etcdConfig.Logger = c.logger
-			etcdConfig.PodNames = f.Updater.Pod.Names
-			etcdConfig.Storage = storageService
-			newProvider, err = etcd.New(etcdConfig)
-			if err != nil {
-				return microerror.MaskAny(err)
-			}
-		default:
-			return microerror.MaskAnyf(invalidConfigError, "unsupported provider kind '%s'", k)
+		bridgeConfig.BridgeName = f.Provider.Bridge.Name
+
+		newProvider, err = bridge.New(bridgeConfig)
+		if err != nil {
+			return microerror.MaskAny(err)
 		}
```

Comment:
I removed the options of configuring other providers because we never used them and we do not plan to use them. When we created the endpoint updater the other providers have been used for testing in environments that are fundamentally different of the environments we have today. This shows that the execution of the endpoint updater is easier and more straight forward. I plan to remove all of the code of the other providers in separate PRs.

Reply:
I didn't go deeply into this particular case, but I agree we should have DI only for testing, unless we have very specific requirement.